components/script/dom/gpudevice.rs

```diff
@@ -523,9 +529,74 @@ impl GPUDeviceMethods for GPUDevice {
                 descriptor.layout.id(),
                 bindings,
             ))
-            .expect("Failed to create WebGPU PipelineLayout");
+            .expect("Failed to create WebGPU BindGroup");
 
         let bind_group = receiver.recv().unwrap();
         GPUBindGroup::new(&self.global(), bind_group, valid)
     }
+
+    /// https://gpuweb.github.io/gpuweb/#dom-gpudevice-createshadermodule
+    fn CreateShaderModule(
+        &self,
+        descriptor: RootedTraceableBox<GPUShaderModuleDescriptor>,
+    ) -> DomRoot<GPUShaderModule> {
+        let (sender, receiver) = ipc::channel().unwrap();
+        let program: Vec<u32> = match &descriptor.code {
+            Uint32Array(program) => program.to_vec(),
+            String(program) => program
+                .as_bytes()
+                .iter()
+                .map(|c| *c as u32)
+                .collect::<Vec<u32>>(),
+        };
+        let id = self
+            .global()
+            .wgpu_create_shader_module_id(self.device.0.backend());
+        self.channel
+            .0
+            .send(WebGPURequest::CreateShaderModule(
+                sender,
+                self.device,
+                id,
+                program,
+            ))
+            .expect("Failed to create WebGPU ShaderModule");
+
+        let shader_module = receiver.recv().unwrap();
+        GPUShaderModule::new(&self.global(), shader_module)
+    }
+
+    /// https://gpuweb.github.io/gpuweb/#dom-gpudevice-createcomputepipeline
+    fn CreateComputePipeline(
+        &self,
+        descriptor: &GPUComputePipelineDescriptor,
+    ) -> DomRoot<GPUComputePipeline> {
+        let pipeline = descriptor.parent.layout.id();
+        let program = descriptor.computeStage.module.id();
+        let entry_point = descriptor
+            .computeStage
+            .entryPoint
+            .as_bytes()
+            .iter()
+            .map(|c| *c as i8)
+            .collect::<Vec<i8>>();
```

Comment:
why isn't this a `String`?

Reply:
You are right! This should be a `String`.